src/PathFactory.js

```diff
@@ -30,7 +32,7 @@ export const defaultHandlers = {
 
   // Add iterable and thenable behavior
   [Symbol.asyncIterator]: getIterator(iteratorHandler),
-  then: iterableToThen(iteratorHandler),
+  then: conditionalHandler(iterableToThen(iteratorHandler), ({ subject }) => !subject),
```

Comment:
I have the feeling this can't be right. `iteratorHandler` first tries `SubjectHandler`, then `ExecuteQueryHandler`; the check is only there for `subject`, and that check will only run on the first segment of the path. I thus think this fail for things like `await user.friends.friends`, since the last segment will not have a `subject`.

Can you confirm?

If I am right, no need to fix this yet, because I'm thinking about other things as well. However, we should wonder why the tests didn't catch this; we might not have enough integration tests.

Reply:
If I understand the code correctly, `await user.friends.friends` will produce something in the form of the following:

```
{
  predicate: 'friends'
  parent: {
    predicate: 'friends'
    parent: { subject: 'user', parent: null }
  }
}
```

As such, for a multi-segment path, the path object will not have a `subject`, and as such will have a `.then()`. However, `await user` will only have something like `{ subject: 'user', parent: null }` in its path, so this will not have a `.then()`.

Is this correct, or am I missing something?